client/ayon_houdini/plugins/publish/collect_componentbuilder_lop.py

```diff
@@ -30,19 +30,22 @@ def process(self, instance):
 
         node = hou.node(instance.data["instance_node"])
 
+        try:
+            node.cook(force=True)  # required to clear existing errors
+        except hou.Error as exc:
+            errors = node.errors()
+            if errors:
+                errors = "\n\n - ".join(errors)
+                raise PublishError(
+                    f"Failed to cook node: '{node.path()}'. Please fix it to proceed.",
+                    detail=f"{exc} \n\n - {errors}"
+                )
+
         # Render the component builder LOPs
         # TODO: Do we want this? or use existing frames? Usually a Collector
         #  should not 'extract' but in this case we need the resulting USD
         #  file.
-        node.cook(force=True)  # required to clear existing errors
         node.parm("execute").pressButton()
```

Comment:
```suggestion
        node.parm("execute").pressButton()
        errors = node.errors()
        if errors:
            for error in errors:
                self.log.error(error)
            raise PublishError(f"Failed to save to disk '{node.path()}'")
```

Reply:
This seems to be better as sometimes the cooking time can be taken very long (Pretty sure it happened in bgeo/vdb not sure about LOP though)

Reply:
The main reason for  `try..except..`  clause was to capture the error which make it show up nicely in the publisher. 
So, will it be the same case for`node.errors()` ?

Reply:
This is the original file from develop without the `node.cook(force=True)` 
![image](https://github.com/user-attachments/assets/530632a2-7e18-4b35-9fea-edc7f3863a67)

Reply:
The issue was originally with the `node.cook()` removed is that if there were errors (e.g. make it break now in your local scene)... then fix your scene so it should render correctly, and then just publish but don't recook the node manually that the original errors would still be listed from `node.errors()`

The comment in code mentions that to "force a cook" purely to clear out EXISTING errors prior to the render.

Reply:
ehm.. I fixed my scene and restarted the publish and it didn't complain. 
![image](https://github.com/user-attachments/assets/3b148d3c-9b93-44f9-ac2c-27dae030122c)

Reply:
If that's the case - let's make a separate PR (sorry, I just want to avoid this PR making more than cosmetic changes) and remove the cooking.

The issue we face initially was that if the ROP or graph had previously errored, that just doing the execute button would fail because it wouldn't resolve that the problem has been 'fixed' since. (It may have been that the Houdini session was set to "Update on mouse up" or whatever - but if we can't reproduce it. Then let's clean it up and remove the cook.

Reply:
https://github.com/ynput/ayon-houdini/pull/193/commits/6513a86567e88c9cd983f9c921c6803b960c34ad